OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -471,6 +471,14 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
     std::vector<QueueSubmission> completed_submissions =
         PullCompletedSubmissions(device, query_pool);
 
+    // We sort the completed submissions by timestamp to avoid sending the capture events out of
+    // order.
+    std::sort(completed_submissions.begin(), completed_submissions.end(),
+              [](const QueueSubmission& left, const QueueSubmission& right) {
+                return left.meta_information.pre_submission_cpu_timestamp <
+                       right.meta_information.pre_submission_cpu_timestamp;
+              });
+
     if (completed_submissions.empty()) {
```

Comment:
One doubt, most likely because I'm missing the full picture: Are all submissions completed in order of their cpu timestamp? Or are all completed submissions obtained only if all the submissions for a `present` have completed? Otherwise, if not all submissions from one `present` are completed, couldn't it happen that we process some that are already completed now and in the following call to this method we process other ones that were previously not ready, but that came before the already processed ones?

Reply:
Submissions on the *same queue*, are guaranteed to be completed in the order as the `vkQueueSubmit` happens on the cpu, i.e. based on this submission_cpu_timestamp. The `present` here is just a random point in time, to check if the submissions are already completed. So if I retrieve a submission here, all the submissions on that queue that happend before, were either already ready in the previous present (and this already sent out), or are part of the result of `PullCompletedSubmission`, in which case the `sort` ensures that we send them out in the correct order.

Reply:
Ah alright, thanks for the explanation. So why does `PullCompletedSubmission` return them not in order? And wouldn't it maybe make more sense for `PullCompletedSubmissions` to perform the `sort` instead of returning an undefined order?

Reply:
Yeah you are right. Maybe this is better.

Reply:
Done.